doc/syntax.md

```diff
@@ -215,45 +187,107 @@ Then `let p: T = a; b` is an expression
 
 Then `let p = a; b` is an expression
 
-## Match statement
+# [Nominal Types](https://en.wikipedia.org/wiki/Nominal_type_system)
 
-If `a` is an expression
+| Type           | Equivalent to    | Description                  | Values                                |
+|----------------|------------------|------------------------------|---------------------------------------|
+| `Option<A>`    | `Either<(), A>`  | Option of type `A`           | `None`, `Some(a)`                     |
+| `bool`         | `Either<(), ()>` | Boolean                      | `false`, `true`                       |
+| `u1`           | `Either<(), ()>` | 1-bit unsigned integer       | `0`, `1`                              |
+| `u2`           | `(u1, u1)`       | 2-bit unsigned integer       | `0`, `1`, `2`, `3`                    |
+| `u4`           | `(u2, u2)`       | 4-bit unsigned integer       | `0`, `1`, ..., `15`                   |
+| `u8`           | `(u4, u4)`       | 8-bit unsigned integer       | `0`, `1`, ..., `255`                  |
+| `u16`          | `(u8, u8)`       | 16-bit unsigned integer      | `0`, `1`, ..., `65535`                |
+| `u32`          | `(u16, u16)`     | 32-bit unsigned integer      | `0`, `1`, ..., `4294967295`           |
+| `u64`          | `(u32, u32)`     | 64-bit unsigned integer      | `0`, `1`, ..., `18446744073709551615` |
+| `u128`         | `(u64, u64)`     | 128-bit unsigned integer     | strings of 32 hex digits              |
+| `u256`         | `(u128, u128)`   | 256-bit unsigned integer     | strings of 64 hex digits              |
 
-If `b` is an expression
+Simfony has familiar types like Boolean and unsigned integers in addition to units, sum and products.
 
-If `c` is an expression
+Simfony's type system is weak in the sense that these nominal types are equivalent to a structural type. For instance, `bool` is equivalent to the sum `Either<(), ()>` of two units. `bool` values are also `Either<(), ()>` values and vice versa. `bool` can be used everywhere `Either<(), ()>` can be used.
```

Comment:
In 6edbc8f809fbb8f43d8ebd86be535cfb2a2e3590:

Lol. Yeah, I think this is okay. The proper way to describe this is that Simfony simply doesn't have nominal types. It just has structural types and you're able to give multiple names to the same types. But because we are aspirationally going to support nominal types, this sort of language ("it has nominal types but the type system is so weak they cannot be observed") is probably fine.

Reply:
You're right, Simfony doesn't have nominal types yet. Simfony has type macros that are resolved to structural types. The expressions that work on type macros are themselves macros. I updated the documentation.